### PR TITLE
Drop support for EOL Python 2.6

### PR DESCRIPTION
**Pull request by hugovk**
pytest itself has already dropped support for Python 2.6 (in https://github.com/pytest-dev/pytest/issues/2812) as it's been EOL for over four years.

This drops support for Python 2.6 and updates some code to make use of new Python 2.7+ features.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
     linting
 
 
 [testenv]
 deps=
     greenlet
-    py26: pytest<3.3
     py{27,34,35,36}: pytest
-    py26: twisted<15.5
     py{27,34,35,36}: twisted
```
Comment (altendky) [inline tox.ini]: For both `pytest` and `twisted` dependencies, remove the `py{27,34,35,36}: `.  It's great I just learned how to make this work in time for py26 support to be removed.  :]

File: README.rst [hunk 2]
```diff
@@ -88,6 +88,12 @@ corotwine work with pytest-twisted::
 That's all.
 
 
+.. |PyPI| image:: https://img.shields.io/pypi/v/pytest-twisted.svg
+    :target: https://pypi.python.org/pypi/pytest-twisted
+
+.. |Pythons| image:: https://img.shields.io/pypi/pyversions/pytest-twisted.svg
+   :alt: supported Python versions
+
```
Comment (altendky) [inline README.rst]: I'm not sure quite what is wrong but these badges don't seem to be loading properly.

https://github.com/hugovk/pytest-twisted/blob/44e308c15510c878b4de2a86e4bd66cbc177550e/README.rst
image

Maybe it's just a temporary issue and not a source bug?  I don't know.

Comment (hugovk) [reply README.rst]: Looks like a temporary problem with shields.io, I saw this on another repo too. And someone just reported this shields.io a few minutes ago:

https://github.com/badges/shields/issues/1568

Would you rather these left out for now, or wait a bit longer?

File: tox.ini (hunk 1 above)
Comment (hugovk) [reply tox.ini]: :)  Done!

File: README.rst (hunk 2 above)
Comment (altendky) [reply README.rst]: No, leave it.  Maybe update all to the `:target:` form though?  I have little experience with rst and wasn't aware of that.  Definitely seems better though.

Comment (hugovk) [reply README.rst]: Done! `:alt:` and `:target:` for all.

**Comment by altendky** [comment]
@hugovk thanks for the tweaks.  :]

**Comment by vtitor** [comment]
@hugovk Awesome, thanks.

**Pull request merged**